packages/tokens/non-fungible/src/extensions/royalties/storage.rs

```diff
@@ -0,0 +1,147 @@
+use soroban_sdk::{contracttype, panic_with_error, Address, Env};
+use stellar_constants::{OWNER_EXTEND_AMOUNT, OWNER_TTL_THRESHOLD};
+
+use crate::{non_fungible::NonFungibleTokenError, Base};
+
+/// Storage container for royalty information
+#[contracttype]
+pub struct RoyaltyInfo {
+    pub receiver: Address,
+    pub basis_points: u32,
+}
+
+/// Storage keys for royalty data
+#[contracttype]
+pub enum RoyaltyStorageKey {
+    DefaultRoyalty,
+    TokenRoyalty(u32),
+}
+
+impl Base {
+    /// Sets the global default royalty information for the entire collection.
+    /// This will be used for all tokens that don't have specific royalty
+    /// information.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `receiver` - The address that should receive royalty payments.
+    /// * `basis_points` - The royalty percentage in basis points (100 = 1%,
+    ///   10000 = 100%).
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::RoyaltyTooHigh`] - If the royalty percentage
+    ///   exceeds the maximum allowed value.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["royalty", receiver: Address]`
```

Comment:
- this event is not emitted
- I'd suggest renaming the 1st topic to `default_royalty` or `collection_royalty`
- Should we exclude `receiver` from topics to be consistent with the event below (or the other way around)